### PR TITLE
fix(dev-guide): updated contributing guideline

### LINKED ISSUE #505: Update Developer Guide's Contribution Guidelines
Our developer guide's Contribution Guidelines need to be updated. It's essential to make it align better with our current practices and create a clearer guide for our community interested in contributing.

https://jitsi.github.io/handbook/docs/dev-guide/dev-guide-contributing/

### PR DESCRIPTION
**Pull request by ilaydadastan**
Fixes #505

### CODE REVIEW

**Comment by ilaydadastan** [comment]
Hi @saghul, @bgrozev & @damencho, I've updated the contributing guidelines located under the development guide in our handbook. However, I noticed that the [Contributing](https://github.com/jitsi/jitsi-meet/blob/master/CONTRIBUTING.md) file in the Jitsi-meet repository is outdated. I propose removing the contents of this file and instead providing a direct link or redirection to the current contributing guidelines in our handbook. This would ensure consistency and accuracy across our documentation. I can do this if it is suitable for you.

File: docs/dev-guide/contributing.md [hunk 1]
```diff
@@ -3,25 +3,49 @@ id: dev-guide-contributing
...
-# Reporting Issues and Asking Questions
+# Reporting Issues and Asking Questions 💭
 
 We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the [issue tracker](https://github.com/jitsi/jitsi-meet/issues).
 
-When opening an issue, please provide as much information as possible. Mention the version of Jitsi Meet, Jicofo and JVB you are using, and explain (as detailed as you can) how the problem can be reproduced.
+Feel free to report ***any bugs, ask for new features or anything else*** you need help with. When opening an issue, please provide as much information as possible.
 
-Please ask any general and implementation specific questions on the [community forum](https://community.jitsi.org/) for support.
+For bugs please follow these steps:
```
Comment (damencho) [inline docs/dev-guide/contributing.md:15]: We ask people to report and discuss problems in the community forum, and when we think that is an issue with the code we ask them to open a ticket in the relevant repo. 

Many times they create tickets which we close as they are in wrong repo, or that is not a code problem, but configuration problems, or those are issues they have with meet.jit.si and so on.
This step is important and should not be removed.

**Comment by damencho** [comment]
All info from that doc is already present in the handbook? If so, yeah go for it.

File: docs/dev-guide/contributing.md (hunk 1 above)
Comment (ilaydadastan) [reply docs/dev-guide/contributing.md:15]: I have this sentence that explains it. But if it is not very clear, I can add back what you sent

Screenshot 2024-05-02 at 14 41 32

Comment (damencho) [reply docs/dev-guide/contributing.md:15]: The first step of reporting an issue is opening a discussion in the forum.

Comment (ilaydadastan) [reply docs/dev-guide/contributing.md:19]: Okay I edited it and made a few more edits

File: docs/dev-guide/contributing.md [hunk 2]
```diff
@@ -3,25 +3,55 @@ id: dev-guide-contributing
...
-We would love to have your help. Before you start working however, please read
-and follow this short guide.
+# 🤝 How to Contribute 
+We greatly appreciate your willingness to contribute ❤️ Before you start working however, please take a moment to read and follow this brief guide.
 
-# Reporting Issues and Asking Questions
+# 📥 Reporting Issues and Asking Questions 
 
-We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the [issue tracker](https://github.com/jitsi/jitsi-meet/issues).
+- We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the [issue tracker](https://github.com/jitsi/jitsi-meet/issues).
```
Comment (bgrozev) [inline docs/dev-guide/contributing.md:11]:
```suggestion
- We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the issue tracker of the appropriate project on GitHub.
```

File: docs/dev-guide/contributing.md [hunk 3]
```diff
@@ -3,25 +3,55 @@ id: dev-guide-contributing
...
+
+- **Expected Behavior:**
+  Describe the expected outcome when using the software.
+
+- **Actual Behavior:**
+  Explain what actually happened, including any error messages.
+
+### 💟 Feature Requests
+
+If you have an idea for a new feature or something you'd like to see improved in the Jitsi, here's how you can let us know:
```
Comment (bgrozev) [inline docs/dev-guide/contributing.md:38]:
```suggestion
If you have an idea for a new feature or something you'd like to see improved in Jitsi, here's how you can let us know:
```

File: docs/dev-guide/contributing.md [hunk 4]
```diff
@@ -185,4 +214,17 @@ npx webpack-bundle-analyzer build/app-stats.json
 
 ### Kotlin
 
-For Kotlin code we use the [standard convention](https://kotlinlang.org/docs/coding-conventions.html) and limit line length to 120 characters. We use `ktlint` to enforce formatting.
+- For Kotlin code we use the [standard convention](https://kotlinlang.org/docs/coding-conventions.html) and limit line length to 120 characters. We use `ktlint` to enforce formatting.
+
+## ✅ Code Reviews
+
+- **Submit Your Contribution:** After completing your work, submit your contribution.
+- **Automatic Assignment:** Your pull request will be automatically assigned to appropriate reviewers based on the affected codebase and expertise required for the changes.
```
Comment (bgrozev) [inline docs/dev-guide/contributing.md:222]: I don't think we have that setup. Unless it's something automagical from github?

File: docs/dev-guide/contributing.md [hunk 5]
```diff
@@ -185,4 +214,17 @@ npx webpack-bundle-analyzer build/app-stats.json
...
+- **Review Process:** Reviewers will carefully examine your code, checking for adherence to coding standards, correctness, performance and potential issues.
+- **Feedback and Iteration:** If any issues or suggestions are identified during the review, you'll receive feedback from the reviewers. Address any comments or concerns raised and make necessary revisions to your code.
+- **Test Your Changes:** Before finalizing your contribution, ensure that any relevant tests are run to verify the correctness and functionality of your changes.
```
Comment (bgrozev) [inline docs/dev-guide/contributing.md:225]: Relevant testing and linting should be done locally before even opening a PR. 

```suggestion
- **Automated tests:** Once the PR is in a good state, a team member will trigger the automated tests. The PR needs to merge cleanly on top of master, and test failures or issues discovered at this stage will need to be addressed before the PR is approved for merging.
```

Comment (ilaydadastan) [reply docs/dev-guide/contributing.md]: Thanks for your comments, all fixed @bgrozev

**Comment by ilaydadastan** [comment]
Hi @damencho & @bgrozev, I've implemented all the changes based on your feedback, will we be able to merge it?

File: docs/dev-guide/contributing.md [hunk 6]
```diff
@@ -3,14 +3,18 @@ id: dev-guide-contributing
...
-# How to Contribute 🤝
-We greatly appreciate your willingness to contribute. Before you start working however, please take a moment to read and follow this brief guide.
+# 🤝 How to Contribute 
+We greatly appreciate your willingness to contribute ❤️ Before you start working however, please take a moment to read and follow this brief guide.
 
-# Reporting Issues and Asking Questions 💭
+# 📥 Reporting Issues and Asking Questions 
 
-We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the [issue tracker](https://github.com/jitsi/jitsi-meet/issues).
+- We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the [issue tracker](https://github.com/jitsi/jitsi-meet/issues).
```
Comment (saghul) [inline docs/dev-guide/contributing.md]: Shall we re-discuss this @damencho ? I think it's not a very natural flow. People find the issue tracker first. Then we send them away, but the issue might come back once confirmed.

That creates some friction.

I think improviing the issue template / questionaire would help, and once an issue is identified as something that should be treated in the forum we can close it as such. Thoughts?

File: docs/dev-guide/contributing.md [hunk 7]
```diff
@@ -208,8 +214,17 @@ npx webpack-bundle-analyzer build/app-stats.json
 
 ### Kotlin
 
-For Kotlin code we use the [standard convention](https://kotlinlang.org/docs/coding-conventions.html) and limit line length to 120 characters. We use `ktlint` to enforce formatting.
+- For Kotlin code we use the [standard convention](https://kotlinlang.org/docs/coding-conventions.html) and limit line length to 120 characters. We use `ktlint` to enforce formatting.
+
+## ✅ Code Reviews
+
+- **Submit Your Contribution:** After completing your work, submit your contribution.
+- **Automatic Assignment:** Your pull request will be automatically assigned to appropriate reviewers based on the affected codebase and expertise required for the changes.
```
Comment (saghul) [inline docs/dev-guide/contributing.md]: This is not the case.

File: docs/dev-guide/contributing.md [hunk 8]
```diff
@@ -208,8 +214,17 @@ npx webpack-bundle-analyzer build/app-stats.json
 
 ### Kotlin
 
-For Kotlin code we use the [standard convention](https://kotlinlang.org/docs/coding-conventions.html) and limit line length to 120 characters. We use `ktlint` to enforce formatting.
+- For Kotlin code we use the [standard convention](https://kotlinlang.org/docs/coding-conventions.html) and limit line length to 120 characters. We use `ktlint` to enforce formatting.
+
+## ✅ Code Reviews
+
+- **Submit Your Contribution:** After completing your work, submit your contribution.
```
Comment (saghul) [inline docs/dev-guide/contributing.md:221]: Let's add some text to also encourage Draft PRs if an approach should be discussed. It's frustrating to finish some work only to scrap it later. Better to open a draft PR early for discussion.

**Review by saghul — COMMENTED** [review]
Left some comments!

File: docs/dev-guide/contributing.md (hunk 7 above)
Comment (ilaydadastan) [reply docs/dev-guide/contributing.md]: I already fixed that

File: docs/dev-guide/contributing.md (hunk 6 above)
Comment (bgrozev) [reply docs/dev-guide/contributing.md]: I suggest moving forward with this PR as it's an improvement and describes the current procedure. Let's re-discuss separately.

Comment (saghul) [reply docs/dev-guide/contributing.md]: Good point. I'll approve as I have no other comments.

**Comment by ilaydadastan** [comment]
Thank you for approving, could we merge it if it’s okay? Thanks @saghul

**Pull request merged**